code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -448,7 +448,9 @@
 	air.adjust_moles(GAS_PLASMA, -2*reaction_efficency)
 
 	//clamps by a minimum amount in the event of an underflow.
-	SSresearch.science_tech.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, clamp((reaction_efficency**2)*BZ_RESEARCH_AMOUNT,0.01,BZ_RESEARCH_MAX_AMOUNT))
+	var/turf/holder_turf = get_turf(holder)
```

Comment:
This causes runtime errors if a reaction happens inside a pipe network, use `get_holder_turf()` instead